packages/salesforce-adapter/src/sfdx_parser/sfdx_dump.ts

```diff
@@ -0,0 +1,166 @@
+
+  return new Set(deletionManifest.components.map(comp => `${comp.type.id}.${comp.fullName}`))
+}
+
+const getComponentsToDelete = async (
+  tree: TreeContainer,
+  pkg: DeployPackage,
+  currentComponents: SourceComponent[],
+): Promise<SourceComponent[]> => {
+  const manifestDeletions = await getManifestComponentsToDelete(tree, pkg)
+  const toDelete = currentComponents
+    // source components only returns top level components, if we want to delete child components (e.g - fields)
+    // we have to explicitly iterate over those as well
+    .flatMap(comp => [comp].concat(comp.getChildren()))
+    .filter(comp => manifestDeletions.has(`${comp.type.id}.${comp.fullName}`))
+  // If a component was deleted, we should also delete all of its child components
+  const childComponentsToDelete = toDelete.flatMap(comp => comp.getChildren())
+  return toDelete.concat(childComponentsToDelete)
+}
+
+const compactPathList = (paths: string[]): string[] => {
+  // Remove paths that are nested under other paths to delete so we don't try to double-delete
+  let lastPath: string | undefined
+  return paths
+    .sort()
+    .map(pathToDelete => {
+      if (lastPath !== undefined && isSubDirectory(pathToDelete, lastPath)) {
+        return undefined
+      }
+      lastPath = pathToDelete
+      return pathToDelete
+    })
+    .filter(values.isDefined)
+}
+
+type DumpElementsToFolderFunc = Required<Adapter>['dumpElementsToFolder']
+export const dumpElementsToFolder: DumpElementsToFolderFunc = async ({ baseDir, changes, elementsSource }) => {
+  const [customObjectInstanceChanges, metadataChanges] = _.partition(changes, isInstanceOfCustomObjectChangeSync)
+
+  const resolvedChanges = await awu(metadataChanges)
+    .map(change => resolveChangeElement(change, getLookUpName))
+    .toArray()
+
+  const fetchProfile = buildFetchProfile({ fetchParams: {} })
+  const filterRunner = filter.filtersRunner(
+    {
+      // TODO: This is a hack that we want to replace with something more type-safe in the near future
+      // for now this works because non of the remote filters actually uses the client on preDeploy
+      client: {} as unknown as SalesforceClient,
+      config: {
+        unsupportedSystemFields: UNSUPPORTED_SYSTEM_FIELDS,
+        systemFields: SYSTEM_FIELDS,
+        fetchProfile,
+        elementsSource,
+        flsProfiles: [],
+      },
+    },
+    allFilters.map(({ creator }) => creator),
+  )
+  await filterRunner.preDeploy(resolvedChanges)
+
+  const pkg = createDeployPackage()
+  const { validChanges, errors } = await validateChanges(resolvedChanges)
+  await Promise.all(
+    validChanges.map(change =>
+      addChangeToPackage(pkg, change, NESTED_METADATA_TYPES, {
+        forceAddToManifest: true,
+      }),
+    ),
+  )
+
+  // Load the components we wish to merge with the current project
+  const buffer = await pkg.getZip()
+  const zipTree = await ZipTreeContainer.create(buffer)
```

Comment:
Since `buffer` is an unhelpful name I would:
```suggestion
  const zipTree = await ZipTreeContainer.create(await pkg.getZip())
```